backend/chembl_drug_search.py

```
@@ -0,0 +1,202 @@
+    if own_conn:
+        conn = sqlite3.connect(DB_PATH)
+    try:
+        cursor = conn.cursor()
+
+        # Find which chembl_ids already exist locally.
+        chembl_ids = [h["chembl_id"] for h in hits if h.get("chembl_id")]
+        existing: Set[str] = set()
+        if chembl_ids:
+            placeholders = ",".join(["?"] * len(chembl_ids))
+            cursor.execute(
+                f"SELECT chembl_id FROM drugs WHERE chembl_id IN ({placeholders})",
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)